### PR TITLE
feat(core): job queue interface

### PR DESCRIPTION
**Pull request by stephhuynh18**
## Description

Minimal job queue implementation and interfaces

Also needed to fix broken existing tests

### CODE REVIEW

File: packages/core/src/sync/sync-api.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  pauseModelSync(models: string | string[]): Promise<void>
+}
+
+interface IpfsService {
+  retrieveFromIPFS(cid: CID | string, path?: string): Promise<any>
+  storeCommit(data: any, streamId?: StreamID): Promise<CID>
+  storeRecord(record: Record<string, unknown>): Promise<CID>
+}
+
+interface Protocol {
```
Comment (stbrody) [inline packages/core/src/sync/sync-api.ts:22]: Protocol is a pretty generic name.  Consider something more descriptive or at least a comment explaining what this represents

File: packages/core/src/sync/sync-api.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  async startModelSync(
+    startBlock: number,
+    endBlock: number,
+    models: string | string[]
+  ): Promise<void> {
+    // add to job queue
+  }
+
+  async pauseModelSync(models: string | string[]): Promise<void> {
```
Comment (stbrody) [inline packages/core/src/sync/sync-api.ts:49]: do we need the ability to pause individual models yet?  Should we have a single shutdown() function that stops all syncs?

**Review by stbrody — APPROVED** [review]
lgtm

File: packages/core/src/indexing/__tests__/build-indexing.test.ts [hunk 3]
```diff
@@ -1,17 +1,30 @@
...
 const diagnosticsLogger = new LoggerProvider().getDiagnosticsLogger()
 // @ts-ignore default import
 const getDatabase = pgTest.default
 
+jest.setTimeout(150000)
+
 describe('sqlite', () => {
   let databaseFolder: tmp.DirectoryResult
 
+  beforeAll(async () => {
```
Comment (PaulLeCam) [inline packages/core/src/indexing/__tests__/build-indexing.test.ts:20]: Why do we need to setup Postgres when running the SQLite tests please?
does this only affect the test setup, or does it become necessary to run Ceramic even with historical indexing disabled?

**Review by PaulLeCam — APPROVED** [review]
Interfaces LGTM, I'm just wondering about the logic to make historical sync Postgres-only, that would be handled at a different level I suppose?

File: packages/core/src/indexing/__tests__/build-indexing.test.ts (hunk 3 above)
Comment (stephhuynh18) [reply packages/core/src/indexing/__tests__/build-indexing.test.ts:20]: oooo good catch! You only need postgres if using postgres =)

File: packages/core/src/sync/sync-api.ts (hunk 2 above)
Comment (stephhuynh18) [reply packages/core/src/sync/sync-api.ts:49]: Yea i don't think we need the ability yet so removed and added shutdown instead.

**Pull request merged**